tests/unit/s2n_client_supported_versions_extension_test.c

```diff
@@ -66,28 +66,23 @@ int main(int argc, char **argv)
 
         int size_result = s2n_extensions_client_supported_versions_size(client_conn);
         EXPECT_NOT_EQUAL(size_result, -1);
-        uint16_t expected_length = (uint16_t) size_result;
+        uint16_t expected_length = size_result - S2N_EXTENSION_TYPE_FIELD_LENGTH - S2N_EXTENSION_LENGTH_FIELD_LENGTH;
```

Comment:
Why is expected_length of type `uint6_t`? If we are being specific, then we should probably type-cast or change the type of `size_result`?

Reply:
size_result can't be uint16_t because it can be negative, because that's how our size functions indicate failure. One of their issues :) But expected_length is uint16_t because it's a two byte size.